### PR TITLE
Completed exercise 4

### PR DESCRIPTION
**Pull request by thischrisblack**

### CODE REVIEW

File: src/exercise/04.html [hunk 1]
```diff
@@ -8,17 +8,120 @@
...
+</script>
+
+<!--EXTRA CREDIT 3-->
+<div id="extra-3"></div>
+<script type="text/babel">
+  const MyPropTypes = {
+    string(props, propName, componentName) {
+      if (typeof props[propName] !== 'string') {
+        return new Error(
+          `you MISTYPE miette? you type her ${propName} like the ${(typeof props[propName]).toUpperCase()}? oh! oh! jail for mother! jail for mother for One Thousand Years!!!!`
```
Comment (jacobschweitzer) [inline src/exercise/04.html:72]: 😆

File: src/exercise/04.html [hunk 2]
```diff
@@ -8,17 +8,120 @@
...
+        {message({children: 'Hello World'})} {message({children: 'Goodbye World'})}
       </div>
     )
 
     // 💯 This is only the first step to making actual React components. The rest is in the extra credit!
     ReactDOM.render(element, document.getElementById('root'))
   </script>
+
+<!--EXTRA CREDIT 1-->
+<div id="extra-1"></div>
```
Comment (jacobschweitzer) [inline src/exercise/04.html:28]: I like that you included all the extra credit separately, I just iterated on the same one.

**Review by jacobschweitzer — APPROVED** [review]
🐴

File: src/exercise/04.html (hunk 2 above)
Comment (thischrisblack) [reply src/exercise/04.html:28]: Glad you like it! I was afraid someone would be like ARGH WHY. But I like to keep all of it for reference.

**Pull request merged**